client.c

```diff
@@ -1372,8 +1378,17 @@ static int fio_client_handle_iolog(struct fio_client *client,
 					pdu->nr_samples * sizeof(struct io_sample));
 		}
 		fclose(f);
-		return 0;
+		ret = 0;
 	}
+
+out:
+	if (pdu)
```

Comment:
Should this be:

if (pdu && pdu != cmd->payload)
        free(pdu);

since pdu is only allocated for the compressed case.

Reply:
Updated this way.